### PR TITLE
[task] Fix Cornell Movies

### LINKED ISSUE #3626: Fail to train any agent with conell_movie dataset
For example, when I train transformer/generator with cornell_movie dataset by command:
`
parlai train_model -t cornell_movie -m transformer/generator -mf ./data/models/transformer_generator/model_cornell_movie_lr1 -gpu 0 -bs 64 -eps 11 -sval True -veps 1 -lr 1
`

The following error occurred:
`
17:54:42 | Current ParlAI commit: 2426d74b93184689be5067bdbf99f1ba96748f7b
17:54:43 | creating task(s): cornell_movie
17:54:43 | loading fbdialog data: /data4/lulc/ben/util/ParlAI/data/CornellMovie/train.txt
17:54:44 | training...
Traceback (most recent call last):  
  File "/data4/lulc/anaconda3/envs/BEN/bin/parlai", line 33, in <module>
    sys.exit(load_entry_point('parlai', 'console_scripts', 'parlai')())
  File "/data4/lulc/ben/util/ParlAI/parlai/__main__.py", line 14, in main    
    superscript_main()
  File "/data4/lulc/ben/util/ParlAI/parlai/core/script.py", line 306, in superscript_main
    return SCRIPT_REGISTRY[cmd].klass._run_from_parser_and_opt(opt, parser)  
  File "/data4/lulc/ben/util/ParlAI/parlai/core/script.py", line 89, in _run_from_parser_and_opt
    return script.run()
  File "/data4/lulc/ben/util/ParlAI/parlai/scripts/train_model.py", line 935, in run    
    return self.train_loop.train()
  File "/data4/lulc/ben/util/ParlAI/parlai/scripts/train_model.py", line 899, in train    
    for _train_log in self.train_steps():
  File "/data4/lulc/ben/util/ParlAI/parlai/scripts/train_model.py", line 802, in train_steps    
    world.parley()
  File "/data4/lulc/ben/util/ParlAI/parlai/core/worlds.py", line 865, in parley
    batch_act = self.batch_act(agent_idx, batch_observations[agent_idx])  
  File "/data4/lulc/ben/util/ParlAI/parlai/core/worlds.py", line 833, in batch_act
    batch_actions = a.batch_act(batch_observation)
  File "/data4/lulc/ben/util/ParlAI/parlai/core/torch_agent.py", line 2131, in batch_act    
    batch = self.batchify(observations)
  File "/data4/lulc/ben/util/ParlAI/parlai/core/torch_generator_agent.py", line 658, in batchify
    batch = super().batchify(obs_batch, sort=sort)  
  File "/data4/lulc/ben/util/ParlAI/parlai/core/torch_agent.py", line 1690, in batchify
    [ex.get('label_truncated_length') for ex in exs]
TypeError: an integer is required (got type NoneType)
`

Actually, any attempts on command line parameters have failed, like '--truncate', '--text-truncate', or '--label-truncate'.
And any agents training with this dataset have failed either.
So, what is the correct way to train cornell_movie dataset?
Thanks~

### PR DESCRIPTION
**Pull request by stephenroller**
**Patch description**
Cornell movies wasn't properly respecting odd/even conversations, and would produce turns with no labels. This patch fixes that. It also takes the opportunity to migrate the teacher to DialogTeacher, away from our deprecated format.

Manually confirmed that the data folds are the same, and the "less data" now was just blank lines.

Also fix a small typo in TA that surfaced the bug in the first place.

Fixes #3626.

**Testing steps**
Updated teacher test, and manual train loop

### CODE REVIEW

File: parlai/tasks/cornell_movie/agents.py [hunk 1]
```diff
@@ -4,109 +4,71 @@
...
+import codecs
 
 
-def _path(opt, filtered):
-    # Build the data if it doesn't exist.
-    build(opt)
-    dt = opt['datatype'].split(':')[0]
-    return os.path.join(opt['datapath'], 'CornellMovie', dt + filtered + '.txt')
+def _path(opt, *additions):
+    return os.path.join(opt['datapath'], 'CornellMovie', *additions)
```
Comment (spencerp) [inline parlai/tasks/cornell_movie/agents.py:17]: Indeed, it seems like `build(opt)` was lost in the change.

File: parlai/tasks/cornell_movie/test/cornell_movie_valid.yml [hunk 2]
```diff
@@ -3,168 +3,28 @@ acts:
...
-    - xxxxxx
-    - yeah, good, ok.
-    - you woke me up to tell me that?
-    - you're gonna like night clubs, they're really a lotta fun.
-    - '{baffled) I don''t follow you.'
-    reward: 0
     text: The "real you".
 num_episodes: 8310
-num_examples: 16759
+num_examples: 13914
```
Comment (spencerp) [inline parlai/tasks/cornell_movie/test/cornell_movie_valid.yml:30]: Oh wow

**Review by spencerp — APPROVED** [review]
Make sure to add `build` back before merging.

Also, have there been any published results that used the duplicate data?

File: parlai/tasks/cornell_movie/agents.py (hunk 1 above)
Comment (stephenroller) [reply parlai/tasks/cornell_movie/agents.py:17]: Oh good catch, thank you!

File: parlai/tasks/cornell_movie/test/cornell_movie_valid.yml (hunk 2 above)
Comment (stephenroller) [reply parlai/tasks/cornell_movie/test/cornell_movie_valid.yml:30]: Ya but the difference didn't have labels, so they weren't being used for metrics anyway.

Comment (spencerp) [reply parlai/tasks/cornell_movie/test/cornell_movie_valid.yml:30]: Ah interesting. So it was just eating compute but not directly affecting the model training. Presumably this was effectively reducing the batch size, then.

**Pull request merged**